build/Targets/Settings.props

```diff
@@ -11,6 +11,7 @@
     <DeveloperBuild Condition="'$(OfficialBuild)' != 'true' AND '$(JenkinsBuild)' != 'true'">true</DeveloperBuild>
     <RepoRoot>$([System.IO.Path]::GetFullPath('$(MSBuildThisFileDirectory)..\..\'))</RepoRoot>
     <NuGetToolPath>$(RepoRoot)nuget.exe</NuGetToolPath>
+    <DisableImplicitNuGetFallbackFolder>true</DisableImplicitNuGetFallbackFolder>
```

Comment:
The tracking bug is https://github.com/dotnet/sdk/issues/1686

For compiler tests, we need to copy a number of ref assemblies into the test output folder (for example `Microsoft.CSharp.dll`). That ref assembly used to come from `.nuget/packages/microsoft.csharp/...`, but now this package isn't even restored because it is in the SDK's fallback folder. Writing logic to locate this ref assembly is difficult, so Nick recommended to disable the fallback folder. 
I don't think there is much of a downside, so I probably would leave as-is for the long-term.

I'll add a comment.